File: src/main/java/com/radixpro/enigma/be/versions/VersionsInjector.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */

package com.radixpro.enigma.be.versions;

import com.radixpro.enigma.AppScope;
import com.radixpro.enigma.xchg.api.XchgApiInjector;

public class VersionsInjector {

   public static Updater injectUpdater(AppScope scope) {
      return new Updater(scope.getAppDb());
   }

   public static AppVersion injectAppVersion(AppScope scope) {
      return new AppVersion(injectUpdater(scope), XchgApiInjector.injectVersionApi(scope));
   }
}

File: src/main/java/com/radixpro/enigma/be/versions/VersionsInjector.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */
package com.radixpro.enigma.be.versions

import com.radixpro.enigma.xchg.api.XchgApiInjector

object VersionsInjector {
    fun injectUpdater(): Updater {
        return Updater()
    }

    @JvmStatic
    fun injectAppVersion(): AppVersion {
        return AppVersion(injectUpdater(), XchgApiInjector.injectVersionApi())
    }
}